src/domain/usecases/GetCoreCompetencyUseCase.ts

```diff
@@ -0,0 +1,11 @@
+import { CoreCompetency } from "$/domain/entities/DataSet";
+import { FutureData } from "$/domain/entities/generic/Future";
+import { CoreCompetencyRepository } from "$/domain/repositories/CoreCompetencyRepository";
+
+export class GetCoreCompetencyUseCase {
```

Comment:
I see that at least this use case has no usage in the app. Typically we won't do that, only create it when known to be used, but of course, you know more context and you may be sure it will be used isolatedly in the UI.